### DISCUSSION TITLE
line clamp bug

### DISCUSSION THREAD

**Discussion by kiranaryal** [tailwindlabs/tailwindcss#13431] [Help]
![image](https://github.com/tailwindlabs/tailwindcss/assets/29688675/297da7be-4d7c-431b-8c22-e339006f0842)

line-clamp-2
after 2 liness there is half line

laravel 10
tailwind ^4
chrome

for eg 
truncate
![Screenshot from 2024-04-02 15-22-53](https://github.com/tailwindlabs/tailwindcss/assets/29688675/7ce771f3-77e7-45c4-83b0-c05b6682f495)

vs  line-clamp-1
![Screenshot from 2024-04-02 15-24-16](https://github.com/tailwindlabs/tailwindcss/assets/29688675/ec269fab-9b4b-4cfe-ae90-05280f5a8751)

supposed to be same thing  right  line-clamp-1  and truncate

**Comment by RobinMalfait** [comment]
Hey!

Can you share a reproduction using https://play.tailwindcss.com?

I think you are using padding on the same element and this is just a strange side effect on how padding interacts with line clamps in CSS. If you move the padding to a parent then it might work.

I'm just guessing, but if you share a reproduction we can try to figure it out!

**Reply by kiranaryal** [reply]
https://play.tailwindcss.com/PfThdvxvMg


why does padding work like this

**Reply by adamwathan** [reply]
Here's the fixed version:

https://play.tailwindcss.com/dF0INJoxrG

Have to put padding on a separate element I'm afraid, just how this works in CSS/browsers, nothing we can control.

**Reply by kiranaryal** [reply]
Thankyou sir,
New thing to learn everyday
Sorry to cause you trouble.

**Reply by Yves852** [reply]
Noticed the class is mainly about hiding overflow.

```css
.line-clamp-2 {
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 2;
}
```

To solve my issue I simply forced the height, no wrapper required. The text isn't actually cropped it's "just under the carpet".
Different styling may require adapted height.

```vue
<div class="line-clamp-2 h-12 text-base">...</div>
```
